src/vm/eventpipefile.cpp

```diff
@@ -19,7 +19,7 @@ EventPipeFile::EventPipeFile(StreamWriter *pStreamWriter) : FastSerializableObje
     }
     CONTRACTL_END;
 
-    m_pBlock = new EventPipeBlock(100 * 1024);
+    m_pBlock = (pStreamWriter->GetType() == StreamWriterType::Ipc ? new EventPipeBlock(1024) : new EventPipeBlock(1024 * 100));
```

Comment:
I looked into this a little more, and I am suspicious this is going to cause you some problems too : ) There is an assumption in the code that an EventBlock is larger than the biggest event and events include stacks. Assuming 8 bytes per stack IP all you would need is a 128 frame stack to exceed the 1KB block size and then you would get an error. At the same time if you wrote very small events, say 20 bytes, once per second you would still have to wait ~50 seconds before they got flushed.

I think what can solve this is to add an API to EventPipeFile that flushes the current block regardless of how full it is. EventPipeFile::WriteEnd() is very close to that, except it also writes the terminator that ends the file.  Once you have that API, update WriteBuffersToFile() to call the new file flush API at the end of its work.